internal/unmarshal.go

```diff
@@ -108,6 +108,20 @@ func IsStruct(typ reflect.Type) bool {
 	return true
 }
 
+func IsOrUnderlyingFeaturesClass(typ reflect.Type) bool {
```

Comment:
why is 'or' in the name?

Reply:
Because it is a features class *or* is an underlying features class but happy to change this if you think it's simpler without the Or

Reply:
I think the `chalkengine` convention is that `is_underlying_x` means `is_x or is a reference to an x`.

Reply:
addressed

Reply:
actually, removed these functions all together, and added `getForeignNamespace` instead